pkg/maps/policymap/policymap.go

```diff
@@ -434,26 +453,55 @@ func (pm *PolicyMap) DumpToSlice() (PolicyEntriesDump, error) {
 		entries = append(entries, eDump)
 	}
 	err := pm.DumpWithCallback(cb)
+	if err != nil {
+		return nil, err
+	}
 
+	// Fetch stats for all dumped entries
+	if pm.stats != nil {
+		for i := range entries {
+			entries[i].Packets, entries[i].Bytes = pm.stats.GetStat(pm.epID, entries[i].Key)
+		}
+	}
 	return entries, err
 }
 
 func (v *PolicyEntry) IsValid(k *PolicyKey) bool {
 	return v.GetPrefixLen() == uint8(k.Prefixlen-StaticPrefixBits)
 }
 
-func newMap(path string) *PolicyMap {
+func newMap(mapPath string) *PolicyMap {
 	mapType := ebpf.LPMTrie
 	flags := bpf.GetPreAllocateMapFlags(mapType)
+
+	// Extract endpoint ID from the given path
+	var id uint16
+	name := path.Base(mapPath)
+	idx := strings.LastIndexByte(name, '_')
+	if idx >= 0 {
+		if id64, err := strconv.ParseUint(name[idx+1:], 10, 16); err == nil {
+			id = uint16(id64)
+		} else {
+			log.WithError(err).Error("failed to parse endpoint ID")
+		}
+	} else {
+		log.WithField(logfields.BPFMapPath, mapPath).Error("malformed policy map name (missing '_')")
+	}
+	statsMap, err := OpenStatsMap()
+	if err != nil {
+		log.WithError(err).Error("Failed to open StatsMap")
```

Comment:
Is it ok to just fallthrough on these error cases and set `epID` to zero?

Reply:
Yes, as the stats map lookup with the zero epID will produce zero results.